src/components/Common/CSVDragAndDrop.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState } from 'react'
+import { formatFileSize, useCSVReader } from 'react-papaparse'
+
+import useFormatMessage from '../../hooks/useFormatMessage'
+import CSV from '../Icon/CSV'
+import CsvRemove from '../Icon/CsvRemove'
+
+import Text from './Typography/Text'
+
+import './CSVDragAndDrop.css'
+
+type CSVError = {
+  type: string
+  code: string
+  message: string
+}
+
+type CSVFile = {
+  data: string[][]
+  errors: CSVError[][]
+  meta: unknown[]
+}
+
+interface Props {
+  onUploadAccepted: (data: string[][]) => void
+  onRemoveFile?: () => void
+}
+
+export default function CSVDragAndDrop({ onUploadAccepted, onRemoveFile }: Props) {
+  const { CSVReader } = useCSVReader()
+  const t = useFormatMessage()
+  const [errorMsg, setErrorMsg] = useState<string | undefined>()
+
+  const fileHandler = (data: CSVFile) => {
+    if (data.errors.length > 0) {
+      setErrorMsg(data.errors[0][0].message)
+    } else {
+      onUploadAccepted(data.data)
+    }
+  }
+
+  return (
+    <CSVReader
+      onUploadAccepted={fileHandler}
+      onDragOver={(event: DragEvent) => {
+        event.preventDefault()
+      }}
+      onDragLeave={(event: DragEvent) => {
+        event.preventDefault()
+      }}
+    >
+      {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ({ getRootProps, acceptedFile, ProgressBar, getRemoveFileProps }: any) => {
```

Comment:
Maybe there is a type you can import from react-papaparse?

Reply:
Not even the hook component is typed 😢
![image](https://github.com/decentraland/governance/assets/45410089/b2516c70-26d4-43de-a331-8d860b4d2eb7)